src/vs/base/common/event.ts

```diff
@@ -111,6 +111,36 @@ export namespace Event {
 		};
 	}
 
+	/**
+	 * Given an event, returns another event which only fires once, and only when the condition is met.
+	 *
+	 * @param event The event source for the new event.
+	 */
+	export function onceIf<T>(event: Event<T>, condition: (e: T) => boolean): Event<T> {
```

Comment:
Maybe I misunderstand but could `Event.once(Event.filter(...))` be used instead?